src/prettier/index.spec.ts

```diff
@@ -43,9 +39,199 @@ describe('@co-it/schematics:prettier', () => {
+  let runner: SchematicTestRunner;
+  let warn: jest.SpyInstance;
+  let project: Tree;
+
+  beforeEach(() => {
+    mockLogger = { warn: () => {} };
+    runner = new SchematicTestRunner('prettier', collectionPath);
+    runner['_logger'] = { createChild: () => mockLogger as any } as any;
+    warn = jest.spyOn(mockLogger, 'warn');
+
+    project = new UnitTestTree(Tree.empty());
+
+    const packageBeforeInstall = {
+      scripts: {},
+      devDependencies: {},
+      prettier: {}
+    };
+    project.create('package.json', JSON.stringify(packageBeforeInstall));
+  });
+
+  it('should warn if configuration is detected in package.json', () => {
+    runner.runSchematic('prettier', {}, project);
+    expect(warn).toHaveBeenCalledWith(
+      'Found competing prettier configuration in package.json.'
+    );
+  });
+
+  it.each([
+    ['.prettierrc.yaml'],
+    ['.prettierrc.yml'],
+    ['.prettierrc.toml'],
+    ['.prettierrc.json'],
+    ['.prettierrc.js'],
+    ['.prettier.config.js']
+  ])(' should warn if configuration is detected in %s', (file: string) => {
+    project.create(file, JSON.stringify({}));
+    runner.runSchematic('prettier', {}, project);
+    expect(warn).toHaveBeenCalledWith(
+      `Found competing prettier configuration in ${file}.`
+    );
+  });
+});
+
+describe('When the developer wants to use a commit hook', () => {
+  let runner: SchematicTestRunner;
+  let project: Tree;
+  let defaultParameters: PrettierSchematicOptions;
+
+  beforeEach(() => {
+    runner = new SchematicTestRunner('prettier', collectionPath);
+    project = new UnitTestTree(Tree.empty());
+    defaultParameters = readParameterDefaults<PrettierSchematicOptions>(
+      parameterSchema
+    );
+
+    const packageBeforeInstall = { scripts: {}, devDependencies: {} };
+    project.create('package.json', JSON.stringify(packageBeforeInstall));
+  });
+
+  it.each([['husky'], ['pretty-quick'], ['lint-staged']])(
+    'should install %s',
+    packageId => {
+      const tree = runner.runSchematic('prettier', defaultParameters, project);
+      const packageJson = JSON.parse(tree.readContent('package.json'));
+
+      expect(packageJson.devDependencies).toEqual(
+        expect.objectContaining({ [packageId]: expect.anything() })
+      );
+    }
+  );
+
+  it('should configure a pre-commit hook in .huskyrc.json', () => {
+    const tree = runner.runSchematic('prettier', defaultParameters, project);
+    const huskyRc = JSON.parse(tree.readContent('.huskyrc.json'));
+
+    expect(huskyRc.hooks['pre-commit']).toEqual(
+      'pretty-quick --staged && lint-staged'
+    );
+  });
+
+  it('should configure lint-staged to check typescript files', () => {
+    const tree = runner.runSchematic('prettier', defaultParameters, project);
+    const lintStagedRc = JSON.parse(tree.readContent('.lintstagedrc'));
+
+    expect(lintStagedRc['*.ts']).toEqual(['tslint --fix', 'git add']);
+  });
+});
+
+describe('When .huskyrc.json already exists', () => {
```

Comment:
We should also test if a husky entry already exists in package.json, like you did for prettier

Reply:
See: https://github.com/co-IT/schematics/pull/11/commits/1d8d57199c75c89bd98f9bbf985bcd2d9fb48aaa